src/WebJobs.Script/Description/DocumentDBBindingMetadata.cs

```diff
@@ -25,9 +22,9 @@ public override void ApplyToConfig(JobHostConfigurationBuilder configBuilder)
             }
 
             DocumentDBConfiguration config = new DocumentDBConfiguration();
-            if (!string.IsNullOrEmpty(ConnectionString))
+            if (!string.IsNullOrEmpty(Connection))
             {
-                config.ConnectionString = ConnectionString;
+                config.ConnectionString = Utility.GetAppSettingOrEnvironmentValue(Connection);
             }
 
```

Comment:
I think you/Mike wrote this code, right? Anyhow, that's a separate issue/bug from this review. Please verify that the updates to your binding that I made make sense. You should investigate this other issue independently :)